src/hotspot/share/prims/jvmtiTagMap.cpp

```diff
@@ -2926,20 +2800,14 @@ void VM_HeapWalkOperation::doit() {
 
   // the heap walk starts with an initial object or the heap roots
   if (initial_object().is_null()) {
-    // If either collect_stack_roots() or collect_simple_roots()
-    // returns false at this point, then there are no mark bits
-    // to reset.
-    ObjectMarker::set_needs_reset(false);
-
-    // Calling collect_stack_roots() before collect_simple_roots()
```

Comment:
This is now done in the Marker constructor, right?

Reply:
> This is now done in the Marker constructor, right?

Yep, exactly. It needed to be reset previously, because it used to be a global variable that would be reused during next heapwalk.